File: Televentas Java/Clases/Catalogo.java
package Clases;

import java.util.ArrayList;
import java.util.List;

public class Catalogo {

    protected List<Producto> productos;

    public Catalogo() {
        this.productos = new ArrayList<>();
    }

    public void listarProductos() {
        for (Producto producto : productos) {
            System.out.println(producto);
        }
    }

    public void agregarProducto(Producto producto) {
        productos.add(producto);
    }

    public Producto buscarProducto(String codigo) {
        for (Producto producto : productos) {
            if (producto.getCodigo().equals(codigo)) {
                return producto;
            }
        }
        return null;
    }

    public void enviarPorCorreo(String correo) {
        System.out.println("El catálogo fue enviado al correo " + correo);
    }
}

File: TeleVentas/Clases/catalogo.py
"""Módulo que define la clase Catalogo."""


class Catalogo:
    """Representa el catálogo de productos de TeleVentas."""

    def __init__(self):
        self._productos = []

    def listar_productos(self):
        for producto in self._productos:
            print(producto)

    def agregar_producto(self, producto):
        self._productos.append(producto)

    def buscar_producto(self, codigo):
        for producto in self._productos:
            if producto.get_codigo() == codigo:
                return producto
        return None

    def enviar_por_correo(self, correo):
        print(f"El catálogo fue enviado al correo {correo}")
